src/gui/syncconflictsmodel.h

```diff
@@ -0,0 +1,130 @@
+ * for more details.
+ */
+
+#ifndef SYNCCONFLICTSMODEL_H
+#define SYNCCONFLICTSMODEL_H
+
+#include "tray/activitydata.h"
+
+#include "conflictsolver.h"
+
+#include <QAbstractListModel>
+#include <QMimeDatabase>
+#include <QLocale>
+
+namespace OCC {
+
+class SyncConflictsModel : public QAbstractListModel
+{
+    Q_OBJECT
+
+    Q_PROPERTY(OCC::ActivityList conflictActivities READ conflictActivities WRITE setConflictActivities NOTIFY conflictActivitiesChanged)
+
+    Q_PROPERTY(bool allExistingsSelected READ allExistingsSelected NOTIFY allExistingsSelectedChanged)
+
+    Q_PROPERTY(bool allConflictingSelected READ allConflictingSelected NOTIFY allConflictingSelectedChanged)
+
+    struct ConflictInfo {
+        QString mExistingFileName;
+        QString mExistingSize;
+        QString mConflictSize;
+        QString mExistingDate;
+        QString mConflictDate;
+        QUrl mExistingPreviewUrl;
+        QUrl mConflictPreviewUrl;
+        bool mExistingSelected = false;
+        bool mConflictSelected = false;
+        QString mExistingFilePath;
+        QString mConflictingFilePath;
+
+        [[nodiscard]] ConflictSolver::Solution solution() const;
+        [[nodiscard]] bool isValid() const;
+    };
+
+public:
+    enum class SyncConflictRoles : int {
+        ExistingFileName = Qt::UserRole,
+        ExistingSize,
+        ConflictSize,
+        ExistingDate,
+        ConflictDate,
+        ExistingSelected,
+        ConflictSelected,
+        ExistingPreviewUrl,
+        ConflictPreviewUrl,
+    };
+
+    Q_ENUM(SyncConflictRoles)
+
+    explicit SyncConflictsModel(QObject *parent = nullptr);
+
+    [[nodiscard]] int rowCount(const QModelIndex &parent = QModelIndex()) const override;
+
+    [[nodiscard]] QVariant data(const QModelIndex &index, int role = Qt::DisplayRole) const override;
+
+    [[nodiscard]] bool setData(const QModelIndex &index, const QVariant &value, int role = Qt::EditRole) override;
+
+    [[nodiscard]] QHash<int,QByteArray> roleNames() const override;
+
+    [[nodiscard]] Qt::ItemFlags flags(const QModelIndex &index) const override;
+
+    [[nodiscard]] OCC::ActivityList conflictActivities() const;
+
+    [[nodiscard]] bool allExistingsSelected() const;
+
+    [[nodiscard]] bool allConflictingSelected() const;
+
+public slots:
+    void setConflictActivities(OCC::ActivityList conflicts);
+
+    void selectAllExisting(bool selected);
+
+    void selectAllConflicting(bool selected);
+
+    void applyResolution();
+
+signals:
+    void conflictActivitiesChanged();
+
+    void allExistingsSelectedChanged();
+
+    void allConflictingSelectedChanged();
+
+private:
+    void updateConflictsData();
+
+    void setExistingSelected(bool value,
+                             const QModelIndex &index,
+                             int role);
+
+    void setConflictingSelected(bool value,
+                                const QModelIndex &index,
+                                int role);
+
+    OCC::ActivityList mData;
```

Comment:
@mgallien why the sudden change of code-style for private members? We have `_memberName` everywhere else, and this got me confused when reviewing the implementaton

Reply:
done